pkg/debug/debug_logger.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2017 The etcd-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debug
+
+import (
+	"os"
+	"path"
+
+	"github.com/coreos/etcd-operator/pkg/util/constants"
+	"github.com/coreos/etcd-operator/pkg/util/k8sutil"
+
+	"github.com/Sirupsen/logrus"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+var (
+	// This flag should be set to enable debug logging
+	DebugEnabled bool
+)
+
+type DebugLogger struct {
+	// regular log to stdout
+	logger *logrus.Entry
+	// log to file for debugging self hosted clusters
+	fileLogger *logrus.Logger
+}
+
+func New(clusterName string) *DebugLogger {
+	if !DebugEnabled {
+		return nil
+	}
+
+	logger := logrus.WithField("pkg", "debug")
+
+	mountPath := path.Join(constants.OperatorRoot, "debug")
```

Comment:
we can remove this. operator does not have to know anything about the host path mount. 

it just logs to what is configured by the deployment when selfhosted-debug-log is enabled.